app/src/benchmark-programs.lisp

```diff
@@ -26,6 +26,29 @@
      (dotimes (i n)
        (format t "MEASURE ~D ro[~D]~%" i i)))))
 
+;;; Benchmark from https://github.com/qulacs/qulacs
+(defun qualcs-program (n)
+  "The qualcs benchmark, specified to be 10 layers of random RX rotations interleaved with 9 layers of neighboring CNOTs, followed by 100 shots.
+
+We are assuming the CNOTs are dense on an even number of qubits."
+  (safely-parse-quil-string
+   (with-output-to-string (*standard-output*)
+     ;; Initial RX layer.
+     (loop :for q :below n :do
+       (format t "RX(~F) ~D~%" (random (* 2 pi)) q))
+     ;; CNOT-RX layers.
+     (loop :repeat 9 :do
+       ;; CNOT's
+       (loop :with qubits := (alexandria:shuffle (alexandria:iota n))
```

Comment:
Isn't it possible to destructure a list over a loop by two entries at a time?

Reply:
yes, `:for .. :on ..`

Reply:
this is uglier because it will give you a "NIL" value as one of the destructured values, so I opt to not do it